webapi/src/test/scala/org/knora/webapi/models/FileModels.scala

```diff
@@ -0,0 +1,1131 @@
+     * @param valueUUID         custom UUID for the value. Optional. Defaults to `None`.
+     *                          If `None`, a UUID will be generated.
+     * @param valueCreationDate custom creation date for the value. Optional.
+     *                          Defaults to `None`. If `None`, the current instant will be used.
+     * @param valuePermissions  custom permissions for the value. Optional. Defaults to `None`.
+     *                          If `None`, the default permissions will be used.
+     * @return a [[CreateDocumentMessage]] containing a [[CreateResourceV2]] as specified by the parameters.
+     */
+    def make(
+      resourceIri: IRI,
+      internalFilename: String,
+      resourceClassIri: SmartIri = OntologyConstants.KnoraApiV2Complex.DocumentRepresentation.toSmartIri,
+      originalFilename: Option[String] = Some("test.pdf"),
+      pageCount: Option[Int] = Some(1),
+      dimX: Option[Int] = Some(100),
+      dimY: Option[Int] = Some(100),
+      label: String = "test document",
+      comment: Option[String] = Some("This is a document"),
+      project: ProjectADM = SharedTestDataADM.anythingProject,
+      permissions: Option[String] = None,
+      valueIRI: Option[SmartIri] = None,
+      valueUUID: Option[UUID] = None,
+      valueCreationDate: Option[Instant] = None,
+      valuePermissions: Option[String] = None
+    ): CreateDocumentMessage = {
+      val valuePropertyIri: SmartIri = OntologyConstants.KnoraApiV2Complex.HasDocumentFileValue.toSmartIri
+      val valueContent = DocumentFileValueContentV2(
+        ontologySchema = ApiV2Complex,
+        fileValue = FileValueV2(
+          internalFilename = internalFilename,
+          internalMimeType = "application/pdf",
+          originalFilename = originalFilename,
+          originalMimeType = Some("application/pdf")
+        ),
+        pageCount = pageCount,
+        dimX = dimX,
+        dimY = dimY,
+        comment = comment
+      )
+      val value = CreateResourceMessage
+        .make(
+          resourceIri = resourceIri,
+          resourceClassIri = resourceClassIri,
+          label = label,
+          valuePropertyIris = List(valuePropertyIri),
+          values = List(
+            List(
+              CreateValueInNewResourceV2(
+                valueContent = valueContent,
+                customValueIri = valueIRI,
+                customValueUUID = valueUUID,
+                customValueCreationDate = valueCreationDate,
+                permissions = valuePermissions
+              )
+            )
+          ),
+          project = project,
+          permissions = permissions
+        )
+        .value
+      new CreateDocumentMessage(value) {}
+    }
+  }
+
+  /**
+   * Case class holding a [[CreateResourceV2]] object.
+   *
+   * @param value the [[CreateResourceV2]] message.
+   */
+  sealed abstract case class CreateImageMessage private (value: CreateResourceV2)
```

Comment:
Since this cannot be used from the outside directly, there is no need to write a scala doc.